File: leet_code_algorithms/reverse_words_in_a_string/Reverse_words_in_a_string.java
package reverse_words_in_a_string;

public class Reverse_words_in_a_string {
	//My Answer
	public String reverseWords(String s) {
		var wordList = s.trim().split("\\s+");
		for (int i = 0; i < wordList.length / 2; i++) {
			var temp = wordList[i];
			wordList[i] = wordList[wordList.length - 1 - i];
			wordList[wordList.length - 1 - i] = temp;
		}
		return String.join(" ", wordList);
	}

	public String usingStringBuilder(String s) {
		String[] words = s.split("\\s+");
		StringBuilder sb = new StringBuilder();
		for (int i = words.length - 1; i >= 0; i--) {
			sb.append(words[i]).append(" ");
		}
		return sb.toString().trim();
	}
}

// https://code.visualstudio.com/docs/java/java-debugging

File: leet_code_algorithms/reverse_words_in_a_string/reverse_words_in_a_string.ts
// My Answer
function reverseWords(s: string): string {
  const wordList = s.trim().split(/\s+/);
  const reversedWordList = wordList.reverse();
  return reversedWordList.join(" ");
}

//run "tsc .\reverse_words_in_a_string.ts"
//and generate reverse_words_in_a_string.js file in the same directory to debug
